terraform/context.go

```diff
@@ -177,7 +177,11 @@ func NewContext(opts *ContextOpts) (*Context, tfdiags.Diagnostics) {
 	log.Printf("[TRACE] terraform.NewContext: loading provider schemas")
 	schemas, err := LoadSchemas(opts.Config, opts.State, components)
 	if err != nil {
-		diags = diags.Append(err)
+		diags = diags.Append(tfdiags.Sourceless(
+			tfdiags.Error,
+			"Could not satisfy plugin requirements",
```

Comment:
I think this is fine, but "Failed to load plugins" or "Could not load plugins" might be reasonable alternatives.

Reply:
I like it! You've given me a lot of great suggestions for logging/error messages etc lately! 
_makes a note to ask @alisdair to write all my logging from now on_